   From: Jan Rekorajski <baggins@sith.mimuw.edu.pl>
   Date: Wed, 24 Oct 2001 18:04:14 +0200

   On Wed, 24 Oct 2001, David S. Miller wrote:
   
   >    From: Jan Rekorajski <baggins@sith.mimuw.edu.pl>
   >    Date: Wed, 24 Oct 2001 16:45:33 +0200
   >    
   >    Speaking of acenic - it's broken in 2.4.13-pre. I have 3c985 and all I
   >    get with 2.4.13-pre is "Firmware NOT running!". After I backed the
   >    changes from -pre patch it started and works fine. Maybe the problem is
   >    I have it in 32bit PCI slot?
   > 
   > Do you have CONFIG_HIGHMEM enabled?  If so, please try with
   > it turned off.
   
   Nope. No HIGHMEM here.

The patch below should cure the problem.

Linus, please apply.

diff -u --recursive --new-file --exclude=CVS --exclude=.cvsignore vanilla/linux/drivers/net/acenic.c linux/drivers/net/acenic.c
--- vanilla/linux/drivers/net/acenic.c	Fri Oct 12 15:35:53 2001
+++ linux/drivers/net/acenic.c	Wed Oct 24 08:32:43 2001
@@ -1051,7 +1051,8 @@
 	struct ace_private *ap;
 	struct ace_regs *regs;
 	struct ace_info *info = NULL;
-	unsigned long tmp_ptr, myjif;
+	u64 tmp_ptr;
+	unsigned long myjif;
 	u32 tig_ver, mac1, mac2, tmp, pci_state;
 	int board_idx, ecode = 0;
 	short i;
-
To unsubscribe from this list: send the line "unsubscribe linux-kernel" in
the body of a message to majordomo@vger.kernel.org
More majordomo info at  http://vger.kernel.org/majordomo-info.html
Please read the FAQ at  http://www.tux.org/lkml/
